app/components/search_input_component.rb

```diff
@@ -11,7 +11,8 @@ def initialize(id: '',
                  item_base_url:,
                  id_key:,
                  links_target: '_top',
-                 search_icon_type: nil)
+                 search_icon_type: nil,
+                 display_mode: 'limited')
```

Comment:
using strings as keys is not a good thing, because you need to remember the options by looking at the code, and you can easily mistype them. Just use a boolean in this case `show_all: false`